torchvision/transforms/functional_tensor.py

```diff
@@ -67,22 +76,51 @@ def crop(img: Tensor, top: int, left: int, height: int, width: int) -> Tensor:
     return img[..., top:top + height, left:left + width]
 
 
-def rgb_to_grayscale(img: Tensor) -> Tensor:
+def rgb_to_grayscale(img: Tensor, num_output_channels: int = 1) -> Tensor:
     """Convert the given RGB Image Tensor to Grayscale.
     For RGB to Grayscale conversion, ITU-R 601-2 luma transform is performed which
     is L = R * 0.2989 + G * 0.5870 + B * 0.1140
 
     Args:
         img (Tensor): Image to be converted to Grayscale in the form [C, H, W].
+        num_output_channels (int): number of channels of the output image. Value can be 1 or 3. Default, 1.
```

Comment:
I wonder if we need this at all or if we should move this logic to the transform itself?

Reply:
Currently, `Grayscale` transform is using `num_output_channels` as argument and `RandomGrayscale` outputs the same number of channels as in the input.
If you are OK with BC change, we can remove it. It is a bit difficult to estimage how many users apply `F.to_grayscale` with this option.